docs/proposals/20201123-tracing-ingestion-api-and-opentelemetry-collector.md

```diff
@@ -0,0 +1,91 @@
+
+## Why
+
+At the moment Observatorium does not expose API to ingest distributed tracing data.
+
+## Goals
+
+* Expose API in the Observatorium API service to ingest tracing data.
+* Deploy a component that receives tracing data and is able to forward it to tracing platforms with storage.
+
+## Non-Goals
+
+* Store data in a persistent storage.
+* Provide read API for tracing data.
+
+## How
+
+The how is split into two sections:
+1. Expose tracing ingestion API
+2. Add deployment manifests for OpenTelemetry collector
+
+### Expose tracing ingestion API in the Observatorium API service
+
+There are a couple of open-source tracing protocols out there - Zipkin, Jaeger and OpenTelemetry. At the moment the OpenTelemetry seems to be the [most popular](https://opentelemetry.io/vendors/), and it is projected to have the biggest adoption in the future. Hence, using OpenTelemetry protocol seems to be the most appropriate choice.
+
+The OpenTelemetry protocol primarily supports gRPC for sending traces (OTLP gRPC) with proto encoding. The HTTP protocol with protobuf encoding is supported as well, but the JSON encoding is still in experimental mode. The majority of users use OTLP gRPC and the HTTP is used in environments where gRPC cannot be used (e.g. mobile clients). Because the Observatorium already supports HTTP and the final state is to support both OTLP HTTP and gRPC, the Observatorium could initially support HTTP and in parallel start working on gRPC.
+
+#### Multitenancy
+
+Multitenancy could be handled the same way as it is handled for metrics and logs. The API service could add an HTTP header or attribute/label to data. This label would then be used in the collector to identify the tenant. Then the collector could route the data to a specific exporter per tenant or export the data to platform/storage by using a single exporter.
+
+### Add deployments manifests for OpenTelemetry collector
+
+OpenTelemetry collector allows receiving, transforming and forwarding data to multiple systems. For instance, it can receive tracing data in Jaeger, Zipkin and OpenTelemetry formats. The transformation allows changing attributes, remove personally identifiable information, derive metrics from traces, perform tail-based sampling and more. After the processing the data can be sent to a single or multiple systems. These set of features allows users to solve variety of use-cases. The Observatorium users deploying on premise would be able to easily forward data to an observability vendor and the same time keep part of the data locally in e.g. Jaeger with low retention or without removing PII data.
+
+The OpenTelemetry collector deployment will be as well needed for sending data to Jaeger. The Jaeger collector does not support receiving OTLP protocol.
```

Comment:
I assume you mean a sidecar deployment here?

Reply:
Can you clarify what you mean by sidecar here? I was imagining it being more of a receiving component (similar to e.g. Thanos receive), i.e. API ingests the tracing data -> it is receive by collector -> collector exports it to a tracing backend.

Reply:
Is there any reason and benefit to deploy it as a sidecar? 

We could deploy it as Jaeger's sidecar - basically making it Jaeger's internal detail.